### PR TITLE
Added source flags in sandbox

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Added new flag `flytesnacks-repo` to mount any repository (In flytesnacks example we need flytesnack code to be inside the sandbox container)
- Added cache in build workflow

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by evalsocket** [comment]
@pmahindrakar-oss I close old pr and created a fresh PR with no code change. You can review it

File: cmd/config/subcommand/sandbox/sandbox_config.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package sandbox
 //go:generate pflags Config --default-var DefaultConfig
 var (
 	DefaultConfig = &Config{
-		Debug: true,
+		Debug: false,
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/sandbox/sandbox_config.go:6]: Why is this changed back to false.

File: cmd/sandbox/sandbox_util_test.go [hunk 2]
```diff
@@ -7,6 +7,10 @@ import (
 	"strings"
 	"testing"
 
+	sandboxConfig "github.com/flyteorg/flytectl/cmd/config/subcommand/sandbox"
+
+	//sandboxConfig "github.com/flyteorg/flytectl/cmd/config/subcommand/sandbox"
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/sandbox_util_test.go:12]: Remove this

File: cmd/sandbox/start.go [hunk 3]
```diff
@@ -17,8 +19,17 @@ const (
...
+	
+Start will run the flyte sandbox cluster inside a docker container and setup the config that is required in a debug mode
+::
+
+ bin/flytectl sandbox start --debug 
+	
+Mount your repository inside sandbox 
+::
 
+ bin/flytectl sandbox start --source=$HOME/flyteorg/flytesnacks
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/start.go:32]: isnt the flag name changed?

File: cmd/sandbox/start.go [hunk 4]
```diff
@@ -49,26 +60,26 @@ func startSandboxCluster(ctx context.Context, args []string, cmdCtx cmdCore.Comm
...
-		fmt.Printf("export KUBECONFIG=%v \n", Kubeconfig)
-		return nil
+	go watchError(cli, ID)
+	if err := readLogs(cli, ID, sandboxConfig.DefaultConfig.Debug); err != nil {
+		return err
 	}
-	fmt.Println("Something goes wrong. We are not able to start sandbox container, Please check your docker client and try again \n", emoji.Rocket)
-	fmt.Printf("error: %v", err)
+
+	fmt.Printf("Add (KUBECONFIG) to your environment variabl \n")
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/start.go:82]: nit : variable

File: cmd/sandbox/sandbox_util.go [hunk 5]
```diff
@@ -87,34 +92,44 @@ func startContainer(cli *client.Client) (string, error) {
...
+			Type:   mount.TypeBind,
+			Source: f.FilePathJoin(f.UserHomeDir(), ".flyte"),
+			Target: K3sDir,
+		},
+		// TODO (Yuvraj) Add flytectl config in sandbox and mount with host file system
+		//{
+		//	Type:   mount.TypeBind,
+		//	Source: f.FilePathJoin(f.UserHomeDir(), ".flyte", "config.yaml"),
+		//	Target: "/.flyte/",
+		//},
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/sandbox_util.go:112]: Keep the TODO in L107 but remove these line from L108 - L112

File: cmd/sandbox/sandbox_util.go [hunk 6]
```diff
@@ -87,34 +92,44 @@ func startContainer(cli *client.Client) (string, error) {
...
-			//	Source: f.FilePathJoin(f.UserHomeDir(), ".flyte", "config.yaml"),
-			//	Target: "/.flyte/",
-			//},
-		},
+		Mounts:       volumes,
 		PortBindings: PortBindings,
 		Privileged:   true,
 	}, nil,
-		nil, SandboxClusterName)
+		nil, FLyteSandboxClusterName)
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/sandbox_util.go:131]: FLyteSandboxClusterName -> flyteSandboxClusterName

File: cmd/config/subcommand/sandbox/sandbox_config.go [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+var (
+	DefaultConfig = &Config{
+		Debug: false,
+	}
+)
+
+// Config
+type Config struct {
+	Debug      bool   `json:"debug" pflag:", Enable debugging"`
+	SnacksRepo string `json:"flytesnacks-path" pflag:", Path of your flytesnacks repository"`
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/sandbox/sandbox_config.go:13]: Avoid hyphens in commands line flags . to be consistent with others use camel case

File: cmd/sandbox/sandbox_util.go [hunk 8]
```diff
@@ -147,14 +163,29 @@ func readLogs(cli *client.Client, id string) error {
...
 	}
 	return nil
 }
+
+func spinLoader() {
+	s := spin.New()
+	s.Set(spin.Spin1)
+	for {
+		fmt.Printf("\r  \033[36mIt will take couple of minutes, We will bring up a fresh flyte cluster %v \033[m %s", emoji.ManTechnologist, s.Next())
+		time.Sleep(100 * time.Millisecond)
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/sandbox_util.go:189]: move to command utils

Comment (evalsocket) [reply cmd/sandbox/sandbox_util.go]: I removed it

File: cmd/config/subcommand/sandbox/sandbox_config.go (hunk 7 above)
Comment (evalsocket) [reply cmd/config/subcommand/sandbox/sandbox_config.go]: i used `flytesnacks`

File: cmd/config/subcommand/sandbox/sandbox_config.go (hunk 1 above)
Comment (evalsocket) [reply cmd/config/subcommand/sandbox/sandbox_config.go]: I removed debug flag. I didn't find any useability if we are using true as the default value. It will become more confusing

**Review by EngHabu — APPROVED** [review]
